src/libs/ontuml2gufo/relation_functions.ts

```diff
@@ -0,0 +1,437 @@
+): Quad[] {
+  const { id, name } = relation;
+  const uri = getURI({
+    id,
+    name,
+    uriFormatBy: options.uriFormatBy,
+    relation,
+  });
+
+  const sourceClass = relation.getSource();
+  const targetClass = relation.getTarget();
+
+  const domainClassUri = getURI({
+    id: sourceClass.id,
+    name: sourceClass.name,
+    uriFormatBy: options.uriFormatBy,
+  });
+  const rangeClassUri = getURI({
+    id: targetClass.id,
+    name: targetClass.name,
+    uriFormatBy: options.uriFormatBy,
+  });
+
+  const quads = [
+    quad(
+      namedNode(`:${uri}`),
+      namedNode('rdf:type'),
+      namedNode('owl:ObjectProperty'),
+    ),
+  ];
+
+  if (domainClassUri && rangeClassUri) {
+    quads.push(
+      quad(
+        namedNode(`:${uri}`),
+        namedNode('rdfs:domain'),
+        namedNode(`:${domainClassUri}`),
+      ),
+    );
+
+    quads.push(
+      quad(
+        namedNode(`:${uri}`),
+        namedNode('rdfs:range'),
+        namedNode(`:${rangeClassUri}`),
+      ),
+    );
+  }
+
+  // add label
+  if (name) {
+    quads.push(
+      quad(namedNode(`:${uri}`), namedNode('rdfs:label'), literal(name)),
+    );
+  } else {
+    quads.push(
+      quad(
+        namedNode(`:${uri}`),
+        namedNode('rdfs:comment'),
+        literal('Relation URI was automatically generated.'),
+      ),
+    );
+  }
+
+  return quads;
+}
+
+/**
+ * Transform relation cardinalities to gUFO
+ */
+function transformRelationCardinalities(
+  writer: N3Writer,
+  relation: IRelation,
+  options: IOntoUML2GUFOOptions,
+): Quad[] {
+  const { properties } = relation;
+
+  const domain = properties[0];
+  const range = properties[1];
+
+  const domainCardinality = domain.cardinality;
+  const rangeCardinality = range.cardinality;
+
+  if (domainCardinality && rangeCardinality) {
+    return [
+      ...transformRelationCardinality({
+        writer,
+        relation,
+        cardinality: rangeCardinality,
+        isDomain: true,
+        options,
+      }),
+      ...transformRelationCardinality({
+        writer,
+        relation,
+        cardinality: domainCardinality,
+        isDomain: false,
+        options,
+      }),
+    ];
+  }
+
+  return [];
+}
+
+/**
+ * Transform relation cardinalities to gUFO
+ */
+function transformRelationCardinality({
+  writer,
+  relation,
+  cardinality,
+  isDomain,
+  options,
+}: {
+  writer: N3Writer;
+  relation: IRelation;
+  cardinality: string;
+  isDomain: boolean;
+  options: IOntoUML2GUFOOptions;
+}): Quad[] {
+  // get domain
+  const classElement = isDomain ? relation.getSource() : relation.getTarget();
+  const classUri = getURI({
+    id: classElement.id,
+    name: classElement.name,
+    uriFormatBy: options.uriFormatBy,
+  });
+
+  const quads = [];
+
+  const lowerboundDomainCardinality = getLowerboundCardinality(cardinality);
+  const upperboundDomainCardinality = getUpperboundCardinality(cardinality);
+  const hasInfiniteCardinality = cardinality.includes('*');
+
```

Comment:
You are missing the "1..*" case here.

Reply:
This generates the following snippet:

```
:Marriage rdf:type owl:Class ;
             rdfs:subClassOf gufo:Relator ,
                             [ rdf:type owl:Restriction ;
                               owl:onProperty :mediatesSpouse ;
                               owl:someValuesFrom :Spouse
                             ] ;
```

Reply:
Notice that there is no need to use owl:minQualifiedCardinality, as owl:someValuesFrom precisely matches the semantics of 1..*

Reply:
Done